src/Cake.SqlServer/SqlServerAliases.cs

```diff
@@ -189,22 +186,46 @@ public static void ExecuteSqlCommand(this ICakeContext context, String connectio
 
 
         /// <summary>
-        /// Reads sql commands from file and executes them.
+        /// Execute any SQL command.
         /// </summary>
         /// <param name="context">The Cake context.</param>
-        /// <param name="connectionString">Connection string where to connect to. Connection script must specify what database to connect to</param>
-        /// <param name="sqlFile">Path to a file with sql commands</param>
+        /// <param name="connection">The connection to use. The connection must be open. See <see cref="OpenSqlConnection"/>.</param>
+        /// <param name="sqlCommands">SQL to be executed</param>
+        /// <example>
+        /// <code>
+        ///     #addin "nuget:?package=Cake.SqlServer"
+        /// 
+        ///     using (var connection = OpenSqlConnection(@"Data Source=(LocalDb)\v12.0;Initial Catalog=MyDatabase"))
+        ///     {
+        ///         ExecuteSqlCommand(connection, "create table [CakeTest].dbo.[CakeTestTable] (id int null)");
+        ///         ExecuteSqlCommand(connection, "...");
+        ///     }
+        /// </code>
+        /// </example>
+        [CakeMethodAlias]
+        public static void ExecuteSqlCommand(this ICakeContext context, SqlConnection connection, string sqlCommands)
+        {
+            Guard.ArgumentIsNotNull(context, nameof(context));
+            Guard.ArgumentIsNotNull(connection, nameof(connection));
+            Guard.ArgumentIsNotNull(sqlCommands, nameof(sqlCommands));
+
+            SqlServerAliasesFacade.ExecuteSqlCommand(context, connection, sqlCommands);
+        }
+
+
+        /// <summary>
+        /// Reads SQL commands from a file and executes them.
+        /// </summary>
+        /// <param name="context">The Cake context.</param>
+        /// <param name="connectionString">The connection string. You may want to specify Initial Catalog. If there are changing parameters, <see cref="SqlConnectionStringBuilder"/> is recommended to escape input.</param>
+        /// <param name="sqlFile">Path to a file with SQL commands.</param>
         /// <example>
         /// <code>
         ///     #addin "nuget:?package=Cake.SqlServer"
         ///
-        ///     Task("ReCreate-Database")
-        ///          .Does(() =>
-        ///          {
-        ///             var connectionString = @"Server=(LocalDb)\v12.0";
-        ///             var sqlFile = "./somePath/MyScript.sql";
-        ///             ExecuteSqlCommand(connectionString, sqlFile);
-        ///         });
+        ///     var connectionString = @"Data Source=(LocalDb)\v12.0;Initial Catalog=MyDatabase";

```

Comment:
Please wrap code sample in `Task("ReCreate-Database").Does(() => {/*sample*/})` - for consistency